tracker/labels.go

```diff
@@ -53,6 +58,16 @@ func (t *myTracker) RenameFromMatches(matches []int, matchinMtx [][]float64, old
 		if newIdx != -1 {
 			if matchinMtx[oldIdx][newIdx] != 0 {
 				if newIdx >= 0 && newIdx < len(newDets) && oldIdx >= 0 && oldIdx < len(oldDets) {
+
+					// If the old one says partial and the new one says full, this is a NEW track
+					if oldDets[oldIdx].detClassification != nil && newDets[newIdx].detClassification != nil {
+						if oldDets[oldIdx].isStable() && oldDets[oldIdx].detClassification.Label() == PartialPizzaLabel &&
+							newDets[newIdx].detClassification.Label() == FullPizzaLabel {
+							// Skipping this one will mean newIdx stays in notUsed, so it will be added as a freshTrack
+							continue
+						}
+					}
+
```

Comment:
This is getting so nested, is it possible to make certain parts of this function their own function, in order to make it more readable?